modules/optimization/include/optimizationreporters/OptimizationReporterBase.h

```diff
@@ -81,6 +117,9 @@ class OptimizationReporterBase : public GeneralReporter
    */
   std::vector<Real> fillParamsVector(std::string type, Real default_value) const;
 
+  /// Sets the initial conditions and bounds right before it is needed.
+  virtual void setICsandBounds(){};
```

Comment:
No we need to call this function in the base class `setInitialCondition` method. We allow any child class to override because they have different ICs and may need to set them differently.